src/components/changeset/changeset.repository.ts

```diff
@@ -27,10 +27,25 @@ export class ChangesetRepository extends DtoRepository(Changeset) {
           .where(not({ prop: hasLabel('modifiedAt') }))
           .return('collect(distinct node) as changed')
       )
+      .subQuery((sub) =>
+        sub
+          .with('changeset')
+          .optionalMatch([
+            node('changeset'),
+            relation('out', '', [], { active: true, deleting: true }),
+            node('deletingNode', 'BaseNode'),
+          ])
+          .optionalMatch([
+            node('changeset', 'Changeset', { id }),
+            relation('out', '', [], { deleting: true }),
+            node('deletedNode', 'Deleted_BaseNode'),
+          ])
```

Comment:
Maybe the BaseNode label constraint is just dropped here and we merge into a single match? Is the `active: true` constraint significant?

Reply:
Do you mean...
```
.optionalMatch([
            node('changeset'),
            relation('out', '', [], { deleting: true }),
            node('node', 'BaseNode'), // How can we merge `BaseNode` and `Deleted_BaseNode` here?
])
```

Reply:
```cql
(changeset)-[{ deleting: true }]->(node)
```

Reply:
I don't think `active: true` is necessary here. Let me test with this one.
```
.optionalMatch([
            node('changeset'),
            relation('out', '', [], { deleting: true }),
            node('node'),
])
```

Reply:
Cool 👍 . I don't think optional is needed either.

Reply:
Correct

Reply:
It works as you expected. 👍